libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RoomMemberMapper.kt

```diff
@@ -24,7 +24,7 @@ import org.matrix.rustcomponents.sdk.RoomMember as RustRoomMember
 
 object RoomMemberMapper {
 
-    fun map(roomMember: RustRoomMember): RoomMember =
+    fun map(roomMember: RustRoomMember): RoomMember = roomMember.use {
```

Comment:
You are not using `it` in the `use` block, which is fine, but could be seen as a bug.
What do other devs think?

Reply:
It's true that we should probably use `it` to make it more explicit, but I don't see much harm in using `roomMember` directly.

Reply:
Even if in this case it appears to be safe, it may lead to bad practice if generalized.
- Not using the lambda's param could break smart casting at times (e.g. when `roomMember` is a nullable function or property accessor or when it is defined in a different module). 
- From the reviewer's PoV it will increase the cognitive load: the bigger the lambda becomes, the more care will be required to keep track of the variables used outside of it.
- On the contrary using `it` enforces locality and makes reasoning with the code simpler (i.e. the reader can be logically sure that `it` will never change throughout the lambda)

Reply:
Not sure it's a problem, sometimes we use this syntax to have a stream instead of some if conditions on the nullability or some predicates. I agree the scope is not exactly the same, so we have to be vigilant, but most of the time there will be no consequence from my point of view.

Sometimes, it's better to rename the `it` variable for clarity, `roomMember.use { roomMember -> ... }` will work in this case but I prefer using the original field instead. `rm` or `member` will introduce new names and more complexity.

For simple cases, using `it` is fine.

To answer the original question, I don't think we need a coding rule, I am for letting the developer make the decision

Reply:
I do agree a rule on this would be a bit too much though.

Reply:
I don't have a super strong opinion on this, but will point out all the other mappers I looked at do it in the same way currently so we'd want to change those if we decide we should use the block parameter everywhere.

I think I slightly prefer this way for this particular case, as you can glance at the function, see the parameter name, then glance down and see how it's used without having to think about the `use` at all (it's basically just a side effect you can ignore for the most part).

Reply:
Consensus seems to be "use `it`, but we don't need it to be a hard rule". I've updated the various mappers that used this pattern to use `it`.